app/lib/service/download_counts/sync_download_counts.dart

```diff
@@ -171,3 +171,9 @@ Future<void> syncDownloadCounts() async {
     }
   }
 }
+
+String formatDateForFileName(DateTime date) {
+  final month = date.month > 9 ? date.month : '0${date.month}';
+  final day = date.day > 9 ? date.day : '0${date.day}';
+  return '${date.year}-$month-${day}T00:00:00Z';
```

Comment:
Have you considered `date.toIso8601String().split('T').first` 

See: [DateTime.toIso8601String()](https://api.dart.dev/stable/3.4.4/dart-core/DateTime/toIso8601String.html).

Or maybe just use [String.padLeft](https://api.dart.dev/stable/3.4.4/dart-core/String/padLeft.html).

```dart
final day = '${date.day}'.padLeft(2, '0');
```

Reply:
Or even just `DateTime.utc(2023, 2, 2).toIso8601String().replaceAll(RegExp(r'\.0+Z$'), 'Z')`

In Dart `toIso8601String` includes ms (and if us is not zero, it also includes us).
So typically, you just want to strip ms and us away.

Reply:
I think `padLeft` is the simplest. Going with that.